lite2/client_test.go

```diff
@@ -940,3 +911,34 @@ func TestClientTrustedValidatorSet(t *testing.T) {
 	assert.NotNil(t, valSet)
 	assert.EqualValues(t, 2, height)
 }
+
+func TestClient_ConcurrentVerifications(t *testing.T) {
+	c, err := NewClient(
+		chainID,
+		trustOptions,
+		fullNode,
+		[]provider.Provider{fullNode},
+		dbs.New(dbm.NewMemDB(), chainID),
+		UpdatePeriod(0),
+		Logger(log.TestingLogger()),
+	)
+	require.NoError(t, err)
+
+	var wg sync.WaitGroup
```

Comment:
can we run this multiple times. from what I'm seeing, this can sometimes fail and sometimes succeed. we want it to fail if there's a race each time

Reply:
do you want me to try append it to the other concurrency test then?

Reply:
no, you can just add a for loop